### PR TITLE
[12.0][MIG] mail_tracking: Adapt Cc to 12.0

### PR DESCRIPTION
**Pull request by Tardo**
Cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you specify the changes respecting 11.0? or even better, put 11.0 commit as is + a commit with the needed changes?

**Comment by Tardo** [comment]
@pedrobaeza done

**Comment by pedrobaeza** [comment]
I don't get why 3 commits now. And please identify the changes with a title like "[MIG] mail_tracking: Adapt CC to v12" or similar

**Comment by Tardo** [comment]
@pedrobaeza done

File: mail_tracking/static/src/js/mail_tracking.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ odoo.define('mail_tracking.partner_tracking', function(require){
 
     var _t = core._t;
 
+<<<<<<< HEAD
```
Comment (pedrobaeza) [inline mail_tracking/static/src/js/mail_tracking.js]: Why this conflict?

File: mail_tracking/static/src/js/mail_tracking.js [hunk 2]
```diff
@@ -40,6 +59,16 @@ odoo.define('mail_tracking.partner_tracking', function(require){
...
         },
 
+        /**
+         * State whether this message contains some email Cc values
+         *
+         * @override
+         * @return {boolean}
+         */
+        hasEmailCc: function () {
+            return !!(this._emailCc && (this._emailCc.length > 0));
```
Comment (yajo) [inline mail_tracking/static/src/js/mail_tracking.js]: Remove some unneeded syntax:

```suggestion
            return this._emailCc && this._emailCc.length > 0;
```

File: mail_tracking/static/src/js/mail_tracking.js [hunk 3]
```diff
@@ -40,6 +59,16 @@ odoo.define('mail_tracking.partner_tracking', function(require){
             return !!(this._partnerTrackings && (this._partnerTrackings.length > 0));
```
Comment (yajo) [inline mail_tracking/static/src/js/mail_tracking.js]: KISS

```suggestion
            return this._partnerTrackings && this._partnerTrackings.length > 0;
```

File: mail_tracking/static/src/js/mail_tracking.js [hunk 4]
```diff
@@ -22,12 +22,31 @@ odoo.define('mail_tracking.partner_tracking', function(require){
...
+         * Email are not Cc.
+         *
+         * @return {boolean}
+         */
+        isEmailCc: function (email) {
+            return false;
+        },
     });
 
     Message.include({
```
Comment (yajo) [inline mail_tracking/static/src/js/mail_tracking.js]: I wonder why can't all this code go into `AbstractMessage`? It seems to include sane defaults that will `return false` when a message has no CC :thinking:

File: mail_tracking/static/src/js/mail_tracking.js [hunk 5]
```diff
@@ -53,6 +82,36 @@ odoo.define('mail_tracking.partner_tracking', function(require){
             }
             return this._partnerTrackings;
         },
+
+        /**
+         * Get the email Cc values of this message
+         * If this message has no email Cc values, returns []
+         *
+         * @override
+         * @return {Object[]}
```
Comment (yajo) [inline mail_tracking/static/src/js/mail_tracking.js]:
```suggestion
         * @return {Array}
```

File: mail_tracking/static/src/js/mail_tracking.js [hunk 6]
```diff
@@ -53,6 +82,36 @@ odoo.define('mail_tracking.partner_tracking', function(require){
             }
             return this._partnerTrackings;
         },
+
+        /**
+         * Get the email Cc values of this message
+         * If this message has no email Cc values, returns []
+         *
+         * @override
```
Comment (yajo) [inline mail_tracking/static/src/js/mail_tracking.js]: This isn't overriding anything:

```suggestion
```

File: mail_tracking/models/mail_mail.py [hunk 7]
```diff
@@ -4,7 +4,6 @@
 import time
 from datetime import datetime
 from email.utils import COMMASPACE
-
```
Comment (yajo) [inline mail_tracking/models/mail_mail.py]: Avoid these kind of diffs please.

**Review by yajo — CHANGES_REQUESTED** [review]

File: mail_tracking/static/src/js/mail_tracking.js (hunk 4 above)
Comment (Tardo) [reply mail_tracking/static/src/js/mail_tracking.js]: To ensure that new implementations doesn't break the xml. I don't see the 'real world' scenario... but perhaps other like reuse AbstractMessage with the same xml.

**Comment by Tardo** [comment]
Changes done!

**Review by yajo — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request closed without merging**